java/com/google/flatbuffers/Table.java

```diff
@@ -69,7 +69,8 @@ protected int __offset(int vtable_offset) {
 
   protected static int __offset(int vtable_offset, int offset, ByteBuffer bb) {
     int vtable = bb.array().length - offset;
-    return bb.getShort(vtable + vtable_offset - bb.getInt(vtable)) + vtable;
+    return bb.getShort(vtable + vtable_offset - bb.getShort(vtable)) != 0  ?
```

Comment:
Can you explain what this does? If `vtable` is pointing to the start of the object, this is probably not working. Why is this code different from the other version of `__offset`?